resources/lib/functions.py

```diff
@@ -588,7 +588,10 @@ def processDirectory(results, progress, params):
         name_format = settings.getSetting(name_format)
 
     dirItems = []
-    result = results.get("Items")
+    try:
+        result = results.get("Items")
```

Comment:
I am not sure what the error is then, results.get("Items") should return None if the key is not there and that is the only issue that this should see.
Its probably cos results is None, test for that first, so this perhaps:
```
if results is None:
   return []
result = results.get("Items")
if result is None:
   result = results
```